api-extractor/src/definitions/ApiItem.ts

```diff
@@ -399,6 +419,150 @@ abstract class ApiItem {
     }
     return false;
   }
+
+  /**
+   * This is called by ApiItems to analyze the types that appear in an expression.  For example,
+   * if a Public API function returns a class that is defined in this package, but not exported,
+   * this is a problem. collectTypeReferences() finds all TypeReference child nodes under the
+   * specified node and analyzes each one.
+   */
+  protected collectTypeReferences(node: ts.Node): void {
+    if (node.kind === ts.SyntaxKind.Block || (node.kind >= 262 && node.kind <= 291)) {
```

Comment:
You can probably use the enum constants instead of "262".  I would add a comment explaining that this is a complete range of items that are all part of the JSDoc.